File: core/src/main/java/com/views/wwlmusic/utils/LWWLMusicIllegal.java
package com.views.wwlmusic.utils;

public class LWWLMusicIllegal {
    public static void check(boolean ok) {
        if (!ok) {
            throw new IllegalArgumentException();
        }
    }

    public static void check(boolean ok, Object object) {
        if (!ok) {
            throw new IllegalArgumentException(String.valueOf(object));
        }
    }

    public static Object check(Object object) {
        if (object != null) {
            return object;
        }
        throw new IllegalArgumentException();
    }

    public static Object check(Object object, Object objectMsg) {
        if (object != null) {
            return object;
        }
        throw new IllegalArgumentException(String.valueOf(objectMsg));
    }
}

File: core/src/main/java/com/loitpcore/views/wwlMusic/utils/LWWLMusicIllegal.kt
package com.loitpcore.views.wwlMusic.utils

object LWWLMusicIllegal {

    fun check(ok: Boolean) {
        require(ok)
    }

    fun check(ok: Boolean, any: Any) {
        require(ok) {
            any.toString()
        }
    }

    fun check(any: Any?): Any {
        if (any != null) {
            return any
        }
        throw IllegalArgumentException()
    }

    fun check(any: Any?, objectMsg: Any): Any {
        if (any != null) {
            return any
        }
        throw IllegalArgumentException(objectMsg.toString())
    }
}
